sdk/core/azure-core/azure/core/pipeline/policies/_utils.py

```diff
@@ -67,3 +68,6 @@ def get_retry_after(response):
             parsed_retry_after = parse_retry_after(retry_after)
             return parsed_retry_after / 1000.0
     return None
+
+def get_domain(url):
+    return str(urlparse(url).netloc).lower()
```

Comment:
We're not doing this on _every_ request are we?

Reply:
Unfortunately, we are.

We have to call it to get the original domain so we can detect there is a domain change.

Reply:
This is unfortunate. What kind of perf analysis has been done on this?
Additionally - if the redirect policy was removed, do we still do this operation?

Reply:
This method is only used in redirect.send. If redirect is removed, this method will not be called.

Reply:
To summarize:

this method is only called if RedirectPolicy is in the pipeline and `permit_redirects` is True.

If we do need to enable redirect and want the security check, my understanding is the cost is not evitable.